### DISCUSSION TITLE
Why does the epsilon function in bc_cylinder.f90 gives half of the geometry ?

### DISCUSSION THREAD

**Discussion by gauravity30** [xcompact3d/Incompact3d#194] [Q&A] [answered]
Recently, I modified the bc_cylinder.f90 file to export the points (xm,ym) where epsilon is 1 to a dat file. Then, I scatter plot the points to visualize the geometry. The points give a half-circle instead of a complete circle (plot attached). But the results of the simulation present a complete circle. Is there the assumption of symmetry in epsilon function about the horizontal diameter.  
![shape](https://github.com/xcompact3d/Incompact3d/assets/53783610/13e5187b-f329-446a-bb88-d2bbb44ecce7)

**Comment by slaizet** [comment, marked as answer]
Hi,
The epsilon function does not make any assumption of symmetry. There might be something wrong with your script to extract the points to visualise the geometry.
Best wishes,
Sylvain
